pull.go

```diff
@@ -90,6 +94,22 @@ func (c *Client) Pull(ctx context.Context, ref string, opts ...RemoteOpt) (_ Ima
 		return nil, err
 	}
 
+	// NOTE(fuweid): unpacker defers blobs download. before create image
+	// record in ImageService, should wait for unpacking(including blobs
+	// download).
+	if pullCtx.Unpack {
+		if unpacks != 0 {
```

Comment:
Thanks a lot! 

Is this checking `unpacks != 0` necessary? When `unpacks == 0`, unpack handler doesn't give us any goroutine so `unpackEg.Wait()` returns immediately.

Reply:
updated with `unpackEg` check. PTAL